### PR TITLE
✨ Onboard wstETH on BNB

### PR DESCRIPTION
**Pull request by MartinGbz**

### CODE REVIEW

**Comment by MartinGbz** [comment]
Got a weird import during the generation. I removed it in this commit: https://github.com/bgd-labs/aave-proposals-v3/pull/507/commits/20e17049cec54d8977dedd64f6c6d309e997c87b

**Comment by MartinGbz** [comment]
Test don't pass because [collector](https://bscscan.com/token/0x26c5e01524d2E6280A48F2c50fF6De7e52E9611C?a=0x25Ec457d1778b0E5316e7f38f3c22baF413F1A8C) has only 0.01 wstETH (instead of 0.04 as you secified in the last [PR](https://github.com/bgd-labs/aave-proposals-v3/pull/501))

I'll check with teammate regarding this collector fund issue.

But even if I put 1e16 (instead of 4e16) for the seed amount, the test still not pass. Do you have an idea?

File: src/20241030_AaveV3BNB_OnboardWstETHToAaveV3OnBNBChain/OnboardWstETHToAaveV3OnBNBChain.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+---
+title: "Onboard wstETH to Aave V3 on BNB Chain"
+author: "ACI"
+discussions: "https://governance.aave.com/t/arfc-onboard-wsteth-to-aave-v3-on-bnb-chain/18501"
+snapshot: "https://snapshot.org/#/aave.eth/proposal/0x752c396a86f1f9b60d3e43b7ed223d9d2f66014e03b6b3eb7ac59e8a169d1fe5"
+---
+
```
Comment (sendra) [inline src/20241030_AaveV3BNB_OnboardWstETHToAaveV3OnBNBChain/OnboardWstETHToAaveV3OnBNBChain.md:7]: missing md text

File: diffs/AaveV3BNB_OnboardWstETHToAaveV3OnBNBChain_20241030_before_AaveV3BNB_OnboardWstETHToAaveV3OnBNBChain_20241030_after.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+| variableDebtTokenName | Aave BNB Smart Chain Variable Debt wstETH |
+| variableDebtTokenSymbol | variableDebtBnbwstETH |
+| virtualAccountingActive | true |
+| virtualBalance | 0.04 wstETH [40000000000000000] |
+| optimalUsageRatio | 45 % |
+| maxVariableBorrowRate | 307 % |
+| baseVariableBorrowRate | 0 % |
+| variableRateSlope1 | 7 % |
+| variableRateSlope2 | 300 % |
+| interestRate | ![ir](/.assets/6582200fbeffb1c0178f8988c1500405e780d82b.svg) |
```
Comment (sendra) [inline diffs/AaveV3BNB_OnboardWstETHToAaveV3OnBNBChain_20241030_before_AaveV3BNB_OnboardWstETHToAaveV3OnBNBChain_20241030_after.md:48]: missing asset file

File: src/20241030_AaveV3BNB_OnboardWstETHToAaveV3OnBNBChain/OnboardWstETHToAaveV3OnBNBChain.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+## Simple Summary
+
+This AIP proposes the onboard of wstETH (wrapped staked ETH) to Aave V3 on BNB Chain to improve Aave’s reach across alternative chains and provide increased choice for Aave users on BNB Chain.
+
+## Motivation
+
+wstETH is one of the most popular collateral and borrow tokens on Aave, with over $4 billion in deposits and $100m in borrows across Aave deployments. The onboarding of wstETH to the Aave V3 on BNB Chain will give Aave users more options for generating staking yield on their collateral and increase dominance of Aave on BNB Chain.
+
+## Specification
+
```
Comment (sendra) [inline src/20241030_AaveV3BNB_OnboardWstETHToAaveV3OnBNBChain/OnboardWstETHToAaveV3OnBNBChain.md:17]: would be good to also add here the eMode table

File: src/20241030_AaveV3BNB_OnboardWstETHToAaveV3OnBNBChain/AaveV3BNB_OnboardWstETHToAaveV3OnBNBChain_20241030.sol [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+ * - Discussion: https://governance.aave.com/t/arfc-onboard-wsteth-to-aave-v3-on-bnb-chain/18501
+ */
+contract AaveV3BNB_OnboardWstETHToAaveV3OnBNBChain_20241030 is AaveV3PayloadBNB {
+  using SafeERC20 for IERC20;
+
+  address public constant wstETH = 0x26c5e01524d2E6280A48F2c50fF6De7e52E9611C;
+  uint256 public constant wstETH_SEED_AMOUNT = 4e16;
+
+  function _postExecute() internal override {
+    IERC20(wstETH).forceApprove(address(AaveV3BNB.POOL), wstETH_SEED_AMOUNT);
```
Comment (sendra) [inline src/20241030_AaveV3BNB_OnboardWstETHToAaveV3OnBNBChain/AaveV3BNB_OnboardWstETHToAaveV3OnBNBChain_20241030.sol:23]: on the other pr https://github.com/bgd-labs/aave-proposals-v3/pull/501/files#diff-8b9426a3cb9f826a6585517b4e18f50b4b0302f890ae08bff85789e2e2d22568R29 you where also adding the emission admin. should this still be added? if so, it should also be added on the md

Comment (MartinGbz) [reply src/20241030_AaveV3BNB_OnboardWstETHToAaveV3OnBNBChain/AaveV3BNB_OnboardWstETHToAaveV3OnBNBChain_20241030.sol:23]: Ah indeed, I missed this when using the generator, I'll fix this, thanks!

Comment (MartinGbz) [reply src/20241030_AaveV3BNB_OnboardWstETHToAaveV3OnBNBChain/AaveV3BNB_OnboardWstETHToAaveV3OnBNBChain_20241030.sol:25]: Now the only diference I see on the _postExecute() of the old PR is this:
`AaveV3BNB.COLLECTOR.transfer(wstETH, address(this), wstETH_SEED_AMOUNT);`
As the [BNB Executor](https://bscscan.com/address/0x9390B1735def18560c509E2d0bc090E9d6BA257a) has already the 4e16 wstETH needed, I suppose I don't need to add this line.

Comment (sendra) [reply src/20241030_AaveV3BNB_OnboardWstETHToAaveV3OnBNBChain/AaveV3BNB_OnboardWstETHToAaveV3OnBNBChain_20241030.sol:25]: yup no need

**Review by sendra — APPROVED** [review]

**Review by pavelvm5 — APPROVED** [review]

**Pull request merged**